component/manager.go

```diff
@@ -80,7 +80,24 @@ func (m *Manager) Start(ctx context.Context) error {
 				return errors.Wrap(err, "Failed to start components.")
 			}
 		} else {
-			log.Warnf("ComponentManager: Component %s has no Stop method", name)
+			log.Warnf("ComponentManager: Component %s has no Start method", name)
+		}
+	}
+	return nil
+}
+
+// Init invokes Init method of all components which implements Initer interface
+func (m *Manager) Init(ctx context.Context) error {
```

Comment:
Why do we need this one?

Reply:
For linking components before they are ready. Refer to documentation in the Initer interface. A concrete example is registering message handlers, which is required to happen before start.